proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
This one is new and wasn't in a previous iteration of the full mutable spec. If query parameters gets a setter, I think it should be documented how this affects the result of getUri(), and whether setUri also updates the query params.